src/sdk/pynni/nni/nas/pytorch/base_mutator.py

```diff
@@ -1,69 +1,51 @@
 import logging
 
-import torch.nn as nn
-
-from nni.nas.pytorch.mutables import Mutable
+from nni.nas.pytorch.mutables import Mutable, MutableScope
+from nni.nas.pytorch.utils import StructuredMutableTreeNode
 
 logger = logging.getLogger(__name__)
 
 
-class BaseMutator(nn.Module):
+class BaseMutator:
     def __init__(self, model):
         super().__init__()
-        self.__dict__["model"] = model
-        self.before_parse_search_space()
-        self._parse_search_space()
-        self.after_parse_search_space()
-
-    def before_parse_search_space(self):
-        pass
-
-    def after_parse_search_space(self):
-        pass
-
-    def _parse_search_space(self):
-        for name, mutable, _ in self.named_mutables(distinct=False):
-            mutable.name = name
-            mutable.set_mutator(self)
-
-    def named_mutables(self, root=None, distinct=True):
-        if root is None:
-            root = self.model
-        # if distinct is true, the method will filter out those with duplicated keys
-        key2module = dict()
-        for name, module in root.named_modules():
+        self.model = model
+        self._structured_mutables = StructuredMutableTreeNode(None)
```

Comment:
do I still have named_mutables?

Reply:
No. It's now called `mutables.traverse()`. I can override `__iter__` to get `for mutable in mutables:`, but with `traverse`, I can do more.